### DISCUSSION TITLE
bin file for the yulc controller revision 2.1

### DISCUSSION THREAD

**Discussion by heidepiek** [ale1800/YULC#5] [Q&A]
Hello alessandro, my  name is harrie.  My pen friend robert (mr blinky man😉  ) drew my attention to this piece of hardware. Love to tinker and wanted to give this project a try. I now saw that the 2.1 was on sale, probably because there is another new revision but that should not spoil the fun. So just ordered one. Hopefully in next week. Am very curious to see how this board works and performs. But have a question, can you also make a bin file of the wled version for the yulc controller?  Have read that you can also upload it on this site (https://aaelectronics-docs.com/), but find it easy to have a bin file at hand.

Regards 

Harrie Bosgraaf 👊 

https://drive.google.com/file/d/1qXTNX87Nszx1rbhTx-xvFB0ykNvezytf/view?usp=sharing

**Comment by ale1800** [comment]
Hello, thanks for your trust! 
I still have to update the images on the website, but you'll receive the v2.2. All the boards I'm shipping are v2.2. [Here](https://aaelectronics-docs.com/documentation/yulc/hw_revisions.html#yulc-v2.2) I'm tracking all the revisions with all the changes, writing also small reports of some tests I made.
Regarding the bin file, you can find it [here](https://aaelectronics-docs.com/documentation/yulc/yulc.html#download-resources), the "Download resources" section.
Thanks again and hope you'll like it! :)

**Comment by robertlipe** [comment]
Hello, my European friends!

I see you've "met." We all share a love of blinkie things and slightly
overlapping technical interests, so I trust we'll all be happy and
productive... and basking bathed In The Glow of the Night
<https://en.wikipedia.org/wiki/Domino_(Genesis_song)>.  Without the
missiles, we hope. :-)

I'll try to not speak for Alessandro, but all the production boards are
2.2. Only a few *Chosen Ones* have 2.1 boards; there just weren't
many made. My V2.1 boards have been my daily drivers now for a few months.
I love the flexibility in options in development and that I can power,
develop, and test all from a single cable. My 2.2 board is still in Lonate
Pozzolo, Italy, presumably awaiting a plane ride from nearby Milano
Malpensa Airport, but you know I'm happy to help as best I can even if I
have to guess a little. (I don't think you'll care about the things that
are different yet, anyway...)

I had not yet met the *.docs website. That's very well done, AA!  I'll send
along some minor nits in the next few days, but find the information well
presented. I love the styling choice in the licensing block that looks like
a Food & Drug Agency inspection label
<https://www.fda.gov/food/nutrition-food-labeling-and-critical-foods/nutrition-facts-label-images-download>,
but that resolution, oh my!

I'll work with Harrie to get the changes I've had in my local NightDriver
tree checked in and into his hands.  For example, I've introduced a concept
of a Controller and a SlowPoll (for checking voltages and blown fuses - I
hadn't thought about temperature, but that should probably go there, too.)
and try to get my analog strip PWM work landed. Because I know he'll be
excited to get our equivlent of Audio Reactive online and because we both
inventory 441 microphones, I never thought about polling temperatureRead,
but that's a good idea, too! We don't have MQTT, but we can certainly "blow
the fuse" and shut down the blinky factory if we see the internal smoke
detector going off. That's a good idea.

Since I suspect Harrie will also want to use his remotes, he and I should
probably pick a a pin on that left connector to recommend for the KY-022
just so we can default that on in our builds of NightDriver. If he wants an
SSD1306 configuration, maybe we should go ahead and recommend pins for
that, too. Not because any of these pins should be better than any other,
but just so there's a configuration that works "out of the box" and so that
the hardware and the software don't BOTH have to hunt for the other.

Oh, I should do something with button0, too.

Once I have everything working well in a branch of my one, THEN I'll
beging the process of arguing with Rutger and Dave about accepting them
into official tree. And if they don't, we'll just live in my tree or we'll
make a YULC version of NightDriver or whatever.


Between us , we should be able to produce something cool that works out of
the box.

Related note, I had to become a contributor to help land some changes to
make it work well, but I'm liking https://github.com/tio/tio

Look forward to working more with you both on this!

RJL

P.S. Harrie: ISTR that the S3 isn't your favorite/default. The first time
you take it out of the box (or when it's being a jerk and you want to
hold its head underwater with an esptool erase_flash or something), *hold
the reset button down when you plug it in*. THEN the device name will
change from /dev/cu.SOMETHINGWITHTHECHIPMACADDRESS to /dev/cu.usbserial1101
(or other physical configuration) When making script files, be prepared for
the tty name to wiggle around depending on how the chip has booted. I can
explain that later with USB trivia if you care.
/dev/cu.usbmodem3030F950AB881 # boot held down. Uses hardware USB stack
/dev/cu.usbmodem1101 # using chips "serial stack"
(I) (loop)(C1) WiFi: WL_CONNECTED, IP: 192.168.2.224, Mem: 172820,
LargestBlk: 155636, PSRAM Free: 1822591/2078447, LED FPS: 30 LED Bright:
9%, LED Watts: 11, CPU: 000%, 007%, FreeDraw: 0.019
*(SlowPoll)(C1) Strand Voltage: 4 VDC.   #I need to fix the rounding. *

**Reply by ale1800** [reply]
Hello Robert!

> I had not yet met the *.docs website. That's very well done, AA!  I'll send
along some minor nits in the next few days, but find the information well
presented. I love the styling choice in the licensing block that looks like
a Food & Drug Agency inspection label
<https://www.fda.gov/food/nutrition-food-labeling-and-critical-foods/nutrition-facts-label-images-download>,
but that resolution, oh my!

😂😂😂 I know that resoultion sucks, but I don't remember where the original file is.

>  I never thought about polling temperatureRead,
but that's a good idea, too!

Yeah, I don't have a temperature sensor on the board, but with the integrated one I can easily estimate the overall board temperature. Doing that I can actually turn off both the mosfet if temperature rises above 70°C. If the fan stops working for some reason, the buck converter (if loaded a lot) can generate a lot of heat. 
At home I'm driving around 60W of FCOB analog strips at 12V from channel 1, and I'm giving a 19V input Power supply from the barrel jack. I don't have the fan and the ESP, after some hours, increase to something like 55/60°C. It's absolutely not bad considering that the ESP32-S3 is already quite warm doing nothing (around 36/37°C) and on the other side of the board the regulator is converting 60W. So a 70°C it's a "sweet spot" to leave freedom without the fan small noise and to keep everything safer. On my setup, if I plug the fan I can go below 40*C without any problem.

> Once I have everything working well in a branch of my one, THEN I'll
beging the process of arguing with Rutger and Dave about accepting them
into official tree. And if they don't, we'll just live in my tree or we'll
make a YULC version of NightDriver or whatever.

This would be great, keep me updated 🤙

**Comment by robertlipe** [comment]
Oh. Now I notice that my reply was public. Ooops. I may go edit some names
out of  that. Can you please do the same for my bit you quoted?
The original looks like it came from:
https://github.com/oshwa/certification-mark-generator#pregenerated

You can customize them at
https://oshwa.github.io/certification-mark-generator/facts
...and get the results in a variety of formats.

Software needs a slightly different label
Foo: GPL v3
Bar: GPL v2-only
Blah: Apache

Crispy, Crunchy, Wholesome Open Source Goodness: *100*%

OK, that last line might be much. Just capturing the styling of the look is
the key. Is that label ubiquitous (enough) around the world that the hook
is recognizable?



I saw the temp thing source for your plugin. It has the hint I need on
finding temps. I can check it in the slow loops where I test the fuse. It's
a great idea. I just needed inspiration. I lack original thoughts. :-)
Mine was to "pull the breaker" and have the FET just power down the strip.
Of course, I can shut down the render copmletely, but running the strims
dimmed lets it fail gracefully while still hinting that something's not
right. I'll do something in that space.
